Automattic/woocommerce-payments
# Add Tracker and tracking

Fixes #74 

#### Changes proposed in this Pull Request

* Adds a tracker API that allows us to track what admin does.
* Add tracking events to our  frontend actions using core's tracking function.

Note we only track those who have opted in.

### Testing instructions

Make sure you have opted into tracking under the WC settings > Advanced > WooCommerce.com:

http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=advanced&section=woocommerce_com

#### A. Test backend fired events

I use error log but, you should be able to follow if you use xDebug.

Edit `woocommerce/includes/tracks/class-wc-tracks-client.php` line 111.
below `$pixel=...` add:

```
error_log( var_export( 'recording pixel', true ) . PHP_EOL . __FILE__  );
error_log( var_export( $pixel, true ) . PHP_EOL . __FILE__  );
```

Now tail the debug log file and do the following actions ( remember to turn on WP_DEBUG and the WP_DEBUG_LOG):

1. Create an order.
1. Go to the admin and do a refund on the order. Select refund via WooCommerce Payments.
1. Submit and check the log for the event that was triggered.

In your log file, you will see in the long URL that it as a parameter `_en=wcpay_edit_order_refund_success`

###### To test failure
Create another order then stop the local server or if you're using the sandbox turn of your proxy. 



##### Testing auth capture

1. In WCPay setting turn on manual capture
1. Create a new order 
1.  Edit the order 
1. Under order actions chooses capture 
1. Click the right arrow button to capture
1. Check your log file for the URL with `_en=wcpay_merchant_captured_auth`


#### B Test js fire events

While logged into admin, open the browse inspector and go to the Network tab. Make sure to preserve the log and then under filter enter `_en=wcadmin_order_edit_refunded`



```
Look out for the _en=wcpay_admin_page_viewed

GET https://pixel.wp.com/t.gif?_en=wcpay_admin_page_viewed&_ts=1591173802612&_via_ua=Mozilla%2F5.0+%28Macintosh%3B+Intel+Mac+OS+X+10_15_4%29+AppleWebKit%2F537.36+%28KHTML%2C+like+Gecko%29+Chrome%2F81.0.4044.138+Safari%2F537.36&_via_ip=172.20.0.1&_lg=en-US%2Cen%3Bq%3D0.9&_dr=http%3A%2F%2Flocalhost%3A8082%2Fwp-admin%2F&_dl=http%3A%2F%2Flocalhost%3A8082%2Fwp-admin%2Fadmin.php%3Fpage%3Dwc-admin%26path%3Dpaymentsdeposits&_ut=wpcom%3Auser_id&_ui=21745020&url=http%3A%2F%2Flocalhost%3A8082&blog_lang=en_US&blog_id=178071066&products_count=18&browser_type=php-agent net::ERR_PROXY_CONNECTION_FAILED
```

-------------------
